common/src/main/java/tachyon/exception/NoForceException.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.exception;
+
+/**
+ * Exception for situations where an operation will not be done unless it is forced through. An
+ * example of this would be deleting a nonempty directory without specifying 'recursive'.
```

Comment:
hmm, that sort of makes it sound like the options themselves are invalid. If we use a really-specific exception name (e.g. `NonEmptyDirectoryDeleteException`, it doesn't add much complexity to deciding which exception to throw because it can be ruled out immediately.